crates/sncast/tests/integration/fee.rs

```diff
@@ -1,247 +1,264 @@
+    let err = args.try_into_fee_settings(None).unwrap_err();
 
     assert_eq!(
-        settings,
-        FeeSettings {
-            max_gas: Some(NonZeroU64::new(1_000_000).unwrap()),
-            max_gas_unit_price: Some(NonZeroU128::new(1_000).unwrap()),
-        }
+        err.to_string(),
+        "Fee estimate must be passed when max_fee is provided"
     );
 }
 
 #[tokio::test]
-async fn test_fee_get_max_fee_with_max_gas_unit_price() {
-    let factory = get_factory().await;
-
+async fn test_all_args_none() {
     let args = FeeArgs {
-        max_fee: Some(Felt::from(MAX_FEE).try_into().unwrap()),
-        max_gas: None,
-        max_gas_unit_price: Some(Felt::from(1_000_u32).try_into().unwrap()),
+        max_fee: None,
+        l1_gas: None,
+        l1_gas_price: None,
+        l2_gas: None,
+        l2_gas_price: None,
+        l1_data_gas: None,
+        l1_data_gas_price: None,
     };
 
-    let settings = args
-        .try_into_fee_settings(factory.provider(), factory.block_id())
-        .await
-        .unwrap();
+    let settings = args.try_into_fee_settings(None).unwrap();
 
     assert_eq!(
         settings,
         FeeSettings {
-            max_gas: Some(NonZeroU64::new(MAX_FEE / 1_000).unwrap()),
-            max_gas_unit_price: Some(NonZeroU128::new(1_000).unwrap()),
+            l1_gas: None,
+            l1_gas_price: None,
+            l2_gas: None,
+            l2_gas_price: None,
+            l1_data_gas: None,
+            l1_data_gas_price: None,
         }
     );
-
-    let FeeSettings {
-        max_gas,
-        max_gas_unit_price,
-    } = settings;
-    let max_gas: u64 = max_gas.unwrap().into();
-    let max_gas_unit_price: u128 = max_gas_unit_price.unwrap().into();
-    assert_eq!(u128::from(max_gas) * max_gas_unit_price, MAX_FEE.into());
 }
 
 #[tokio::test]
-async fn test_fee_get_none() {
-    let factory = get_factory().await;
-
-    let args = FeeArgs {
-        max_fee: None,
-        max_gas: None,
-        max_gas_unit_price: None,
+async fn test_from_fee_estimate() {
+    let mock_fee_estimate = FeeEstimate {
+        l1_gas_consumed: Felt::from(1),
+        l1_gas_price: Felt::from(2),
+        l2_gas_consumed: Felt::from(3),
+        l2_gas_price: Felt::from(4),
+        l1_data_gas_consumed: Felt::from(5),
+        l1_data_gas_price: Felt::from(6),
+        unit: PriceUnit::Fri,
+        overall_fee: Felt::from(44),
     };
-
-    let settings = args
-        .try_into_fee_settings(factory.provider(), factory.block_id())
-        .await
-        .unwrap();
+    let settings = FeeSettings::try_from(mock_fee_estimate).unwrap();
 
     assert_eq!(
         settings,
         FeeSettings {
-            max_gas: None,
-            max_gas_unit_price: None,
+            l1_gas: Some(1),
+            l1_gas_price: Some(2),
+            l2_gas: Some(3),
+            l2_gas_price: Some(4),
+            l1_data_gas: Some(5),
+            l1_data_gas_price: Some(6),
         }
     );
```

Comment:
This is basically a unit test, please place it directly in fee.rs file.

Maybe some other from here can be moved as well?

Reply:
Good catch, moved.

> Maybe some other from here can be moved as well?

No, nothing apart of that.